go/runtime/bundle/manager.go

```diff
@@ -236,20 +246,46 @@ func (m *Manager) Queue(runtimeID common.Namespace, manifestHashes []hash.Hash)
 	}
 	m.downloadQueue[runtimeID] = hashes
 
-	// Trigger immediate download of new bundles.
+	// Trigger immediate download and clean-up of bundles.
+	select {
+	case m.triggerCh <- struct{}{}:
+	default:
+	}
+}
+
+// Cleanup updates the runtime's maximum bundle version for pending clean-up.
+//
+// Existing maximum version in the clean-up queue for the specified runtime (if any),
+// is removed and replaced with the given one.
+//
+// Warning: If clean-up fails it's not retried.
+func (m *Manager) Cleanup(runtimeID common.Namespace, version version.Version) {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+
+	m.cleanupQueue[runtimeID] = version
+
+	// Trigger immediate download and clean-up of bundles.
 	select {
-	case m.downloadCh <- struct{}{}:
+	case m.triggerCh <- struct{}{}:
 	default:
 	}
 }
 
-// Download tries to download bundles in the queue.
-func (m *Manager) Download() {
+func (m *Manager) download() {
+	m.logger.Info("downloading bundles")
 	for runtimeID := range m.runtimeIDs {
 		m.downloadBundles(runtimeID)
 	}
 }
 
+func (m *Manager) clean() {
```

Comment:
I would move this before `cleanBundles`, so that you don't need to jump when reading the code. I would also move `cleanBundles` so that the code is before/after `cleanBundlesOnStartup`.

Usually, I want to have public methods on the top, and related methods together.